aptos-move/framework/aptos-framework/sources/resource_account.move

```diff
@@ -58,7 +86,60 @@ module aptos_framework::resource_account {
         optional_auth_key: vector<u8>,
     ) acquires Container {
         let (resource, resource_signer_cap) = account::create_resource_account(origin, seed);
+        rotate_account_authentication_key_and_store_capability(
+            origin,
+            resource,
+            resource_signer_cap,
+            optional_auth_key,
+        );
+    }
 
+    /// Creates a new resource account, transfer the amount of coins from the origin to the resource
+    /// account, and rotates the authentication key to either the optional auth key if it is
+    /// non-empty (though auth keys are 32-bytes) or the source accounts current auth key. Note,
+    /// this function adds additional resource ownership to the resource account and should only be
+    /// used for resource accounts that need access to Coin<AptosCoin>.
+    public entry fun create_resource_account_and_fund(
+        origin: &signer,
+        seed: vector<u8>,
+        optional_auth_key: vector<u8>,
+        fund_amount: u64,
+    ) acquires Container {
+        let (resource, resource_signer_cap) = account::create_resource_account(origin, seed);
+        coin::register<AptosCoin>(&resource);
+        coin::transfer<AptosCoin>(origin, signer::address_of(&resource), fund_amount);
+        rotate_account_authentication_key_and_store_capability(
+            origin,
+            resource,
+            resource_signer_cap,
+            optional_auth_key,
+        );
+    }
+
+    /// Creates a new resource account, publishes the package under this account transaction under
+    /// this account and leaves the signer cap readily available for pickup.
```

Comment:
thank you for adding a function for this! I haven't been able to find any examples showing how to properly serialize package bytecode as an entry function argument (e.g. for `publish_package_txn`). Any chance there is an example that would show how to do so using the typescript sdk? thanks much!

Reply:
Take a look here: https://github.com/aptos-labs/aptos-core/pull/3612